src/bin/inx-chronicle/api/stardust/analytics/routes.rs

```diff
@@ -114,6 +116,15 @@ async fn unspent_output_ledger_analytics<O: OutputKind>(
     database: Extension<MongoDb>,
     LedgerIndex { ledger_index }: LedgerIndex,
 ) -> ApiResult<OutputAnalyticsResponse> {
+    let ledger_index = if let Some(ledger_index) = ledger_index {
+        ledger_index
+    } else {
+        database
+            .collection::<MilestoneCollection>()
+            .get_ledger_index()
+            .await?
+            .ok_or(ApiError::NoResults)?
+    };
```

Comment:
Can we move this to a separate function, like `async fn or_newest(db: &MongoDb, ledger_index: Option<MilestoneIndex>`?

Reply:
Honestly, I don't think that would help at all.

Reply:
I had it like that when I experimented with it and I think it really helped with descriptiveness. It also removes some of the redundancy in the code.